apps/web/src/lib/auth/getCurrentUser.ts

```diff
@@ -0,0 +1,21 @@
+import { cache } from 'react'
+
+import db from '$/db/database'
+import { getCurrentUserFromDB } from '$core/data-access'
+import { Result } from '$core/lib'
+import { getSession } from './getSession'
+
+export const getCurrentUser = cache(async () => {
```

Comment:
React [cache](https://react.dev/reference/react/cache) is like in Rails something that lives through the whole request. So if this method is called 8 times is only executed once